node/src/neighborhood/mod.rs

```diff
@@ -3064,6 +3148,267 @@ mod tests {
+                assert_eq!(
+                    neighborhood
+                        .neighborhood_database
+                        .node_by_key(&t_public_key)
+                        .unwrap()
+                        .metadata
+                        .country_undesirability,
+                    1_000_000u32,
+                    "We expecting 1 000 000, country is not requested for exit location"
+                );
+            }),
+        };
+        let assertion_msg_neighborhood = AssertionsMessage {
+            assertions: Box::new(move |neighborhood: &mut Neighborhood| {
+                assert_eq!(
+                    neighborhood.exit_locations_opt,
+                    Some(HashMap::from([
+                        (1usize, vec!["CZ".to_string()]),
+                        (2usize, vec!["FR".to_string()]),
+                    ]))
+                );
+                assert_eq!(neighborhood.fallback_routing, false);
+            }),
+        };
+        let request_2 = UiSetExitLocationRequest {
+            fallback_routing: true,
+            exit_locations: vec![],
+        };
+        let message_2 = NodeFromUiMessage {
+            client_id: 0,
+            body: request_2.tmb(0),
+        };
+        let assertion_msg_2 = AssertionsMessage {
+            assertions: Box::new(move |neighborhood: &mut Neighborhood| {
+                assert_eq!(
+                    neighborhood
+                        .neighborhood_database
+                        .node_by_key(&q_public_key_2)
+                        .unwrap()
+                        .metadata
+                        .country_undesirability,
+                    0u32,
+                    "We expecting zero, exit_location was unset"
+                );
+                assert_eq!(
+                    neighborhood
+                        .neighborhood_database
+                        .node_by_key(&r_public_key_2)
+                        .unwrap()
+                        .metadata
+                        .country_undesirability,
+                    0u32,
+                    "We expecting zero, exit_location was unset"
+                );
+                assert_eq!(
+                    neighborhood
+                        .neighborhood_database
+                        .node_by_key(&s_public_key_2)
+                        .unwrap()
+                        .metadata
+                        .country_undesirability,
+                    0u32,
+                    "We expecting zero, exit_location was unset"
+                );
+                assert_eq!(
+                    neighborhood
+                        .neighborhood_database
+                        .node_by_key(&t_public_key_2)
+                        .unwrap()
+                        .metadata
+                        .country_undesirability,
+                    0u32,
+                    "We expecting zero, exit_location was unset"
+                );
+            }),
+        };
```

Comment:
Do we need another assertion message here to make sure the field of the Neighborhood is properly reset? Or perhaps that could be part of the `AssertionMessage` above.